cert/aziot-certd/src/lib.rs

```diff
@@ -384,27 +384,74 @@ async fn create_cert_inner<'a>(
         builder.set_pubkey(pubkey)?;
         builder.set_not_before(&*Asn1Time::days_from_now(0)?)?;
 
+        let (subject_name, expiry, issuer_ref) =
+            stack.get(0).map_or((subject_name, expiry, None), |x509| {
+                let issuer_expiry = x509.not_after();
+
+                (
+                    x509.subject_name(),
+                    if expiry < issuer_expiry {
+                        expiry
+                    } else {
+                        issuer_expiry
+                    },
+                    Some(x509.as_ref()),
+                )
+            });
+
         // x509_req.extensions() returns an Err variant if no extensions are
         // present in the req. Ignore this Err and only copy extensions if
         // provided in the req.
         if let Ok(exts) = req.extensions() {
             for ext in exts {
+                // Add the Subject Key ID extension to CA certs.
+                if let Some(basic_constraints) =
+                    openssl2::extension::BasicConstraints::from_ext(&ext)
+                {
+                    if basic_constraints.ca {
+                        let subj_key_id = openssl::x509::extension::SubjectKeyIdentifier::new();
+                        let context = builder.x509v3_context(issuer_ref, None);
+                        let subj_key_id = subj_key_id.build(&context)?;
+                        builder.append_extension(subj_key_id)?;
+                    }
+                }
+
                 builder.append_extension(ext)?;
```

Comment:
@arsing If a CSR already sets the SKID extension (not sure they would be able to set AKID as they probably don't have the issuer key), this line would append the SKID extension twice. Should we check to make sure SKID is only appended once? Right now it doesn't matter because none of the other services generate CSRs with SKID.

Reply:
Yes, we should only add our own SKID extension if there isn't one already in `req`.

Reply:
Fixed.